### PR TITLE
style: :arrow_up: Update pydantic to 1.10.4

### PR DESCRIPTION
**Pull request by chadell**

### CODE REVIEW

**Review by jvanderaa — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -24,8 +24,7 @@ include = [
 [tool.poetry.dependencies]
 python = "^3.7"
 click = ">=7.1, <9.0"
-# Skipping 1.9.x versions due issue https://github.com/samuelcolvin/pydantic/issues/4114
-pydantic = {version = "~1.8.2", extras = ["dotenv"]}
+pydantic = {version = ">= 1.8.0, != 1.9.*, >= 1.10.4", extras = ["dotenv"]}
```
Comment (glennmatthews) [inline pyproject.toml:27]: Remind me, does this actually allow both 1.8.x and 1.10.4+? Or would it be equivalent and simpler to just have `">= 1.10.4"` alone?

**Review by glennmatthews — APPROVED** [review]
Hopefully the tests passing means that samuelcolvin/pydantic#4114 isn't an issue for us any more. :)

File: pyproject.toml (hunk 1 above)
Comment (chadell) [reply pyproject.toml:27]: the idea is to keep the parser working for environments using Pydantic pinned in 1.8.X versions, and start using the newer versions

Comment (glennmatthews) [reply pyproject.toml:27]: Right, just wanted to make sure that's what it's actually doing :)

**Pull request merged**